services/wallet/collectibles/commands.go

```diff
@@ -241,42 +260,75 @@ func (c *loadOwnedCollectiblesCommand) Run(parent context.Context) (err error) {
 			// Normally, update the DB once we've finished fetching
 			// If this is the first fetch, make partial updates to the client to get a better UX
 			if initialFetch || finished {
-				receivedIDs, err := c.ownershipDB.GetIDsNotInDB(c.chainID, c.account, c.partialOwnership)
-				if err != nil {
-					log.Error("failed GetIDsNotInDB in processOwnedIDs", "chain", c.chainID, "account", c.account, "error", err)
-					return err
-				}
-
 				// Token balances should come from the providers. For now we assume all balances are 1, which
 				// is only valid for ERC721.
 				// TODO (#13025): Fetch balances from the providers.
 				balances := ownedTokensToTokenBalancesPerContractAddress(c.partialOwnership)
 
-				err = c.ownershipDB.Update(c.chainID, c.account, balances, start.Unix())
+				updateMessage.Removed, updateMessage.Updated, updateMessage.Added, err = c.ownershipDB.Update(c.chainID, c.account, balances, start.Unix())
 				if err != nil {
 					log.Error("failed updating ownershipDB in loadOwnedCollectiblesCommand", "chain", c.chainID, "account", c.account, "error", err)
 					c.err = err
 				}
 
-				c.receivedCollectiblesCh <- OwnedCollectibles{
-					chainID: c.chainID,
-					account: c.account,
-					ids:     receivedIDs,
+				if len(updateMessage.Removed) > 0 {
+					c.ownedCollectiblesChangeCb <- OwnedCollectiblesChange{
+						ownedCollectibles: OwnedCollectibles{
+							chainID: c.chainID,
+							account: c.account,
+							ids:     updateMessage.Removed,
+						},
+						changeType: OwnedCollectiblesChangeTypeRemoved,
+					}
+				}
+
+				if len(updateMessage.Updated) > 0 {
+					c.ownedCollectiblesChangeCb <- OwnedCollectiblesChange{
+						ownedCollectibles: OwnedCollectibles{
+							chainID: c.chainID,
+							account: c.account,
+							ids:     updateMessage.Updated,
+						},
+						changeType: OwnedCollectiblesChangeTypeUpdated,
+					}
+				}
+
+				if len(updateMessage.Added) > 0 {
+					c.ownedCollectiblesChangeCb <- OwnedCollectiblesChange{
+						ownedCollectibles: OwnedCollectibles{
+							chainID: c.chainID,
+							account: c.account,
+							ids:     updateMessage.Added,
+						},
+						changeType: OwnedCollectiblesChangeTypeAdded,
+					}
 				}
 			}
 
 			if finished || c.err != nil {
 				break
 			} else if initialFetch {
-				c.triggerEvent(EventCollectiblesOwnershipUpdatePartial, c.chainID, c.account, "")
+				encodedMessage, err := json.Marshal(updateMessage)
+				if err != nil {
+					c.err = err
+					break
+				}
+				c.triggerEvent(EventCollectiblesOwnershipUpdatePartial, c.chainID, c.account, string(encodedMessage))
+
+				updateMessage = OwnershipUpdateMessage{}
 			}
 		}
 	}
 
+	encodedMessage, err := json.Marshal(updateMessage)
```

Comment:
I still want the error event to be emitted if encoding fails for some reason, but I added a check to prevent an unnecessary call to `Marshall` in that condition.

Also, this is not done before on the `initialFetch` branch since it breaks out of the loop before that due to the `finished` condition.